Automattic/newspack-network
# Experimental auditing features

### All Submissions:






### Changes proposed in this Pull Request:



A few auditing features, for now restricted to a specific WP user. 

1. In the Membership Plans view, the tables are now unified. Two columns are added to help with auditing the synchronisation:

<img width="1214" alt="image" src="https://github.com/Automattic/newspack-network/assets/7383192/3249809c-e798-4867-b7cc-19d9b3848d03">

2. New column in the WP Users view, featuring origin site URL and User ID on that site (the text is a link to the user edit screen):

<img width="1974" alt="image" src="https://github.com/Automattic/newspack-network/assets/7383192/4ce3bbc5-af95-4584-b871-3ef29d8f9246">

4. New column in the Nodes table, displaying synchronisable users count:

<img width="1122" alt="image" src="https://github.com/Automattic/newspack-network/assets/7383192/cf1abd07-c742-4e20-b6b3-7e63875d4149">

### How to test the changes in this Pull Request:

1. Verify none of the features are visible until the next step is done
2. Set `NEWSPACK_NETWORK_EXPERIMENTAL_AUDITING_USER` wp-config variable, with the value of the admin user you're using (e.g. `admin`)
1. Visit **WP Users** tables (`/wp-admin?users.php`) on the Hub and Node sites, verify the new column features correct data and links
1. Visit the **"Nodes" view** on the Hub site
    1. Click the user counts, observe the links point to WP Users table filtered by synchronized roles
    1. Visit the corresponding Node sites, confirm the numbers are correct
1. Visit the **"Membership Plans" view** on the Hub site 
    1. observe a new "Active Members" column
    1. Observe a new "Discrepancies" column and verify the discrepancies between synchronised plan members*
    2. Click the link to view the discrepant members

\* to create some discrepancies, add plan members and then quickly remove the corresponding event (`wp_newspack_hub_event_log` table) or webhook (`np_webhook_request` post type), if on hub or node, respectively

### Other information:
